libraries/SITL/SIM_Submarine.cpp

```diff
@@ -70,20 +70,39 @@ void Submarine::calculate_forces(const struct sitl_input &input, Vector3f &rot_a
     	body_accel.z = -GRAVITY_MSS;
     }
 
-    float terminal_rotation_rate = 10.0;
-    if (terminal_rotation_rate > 0) {
-        // rotational air resistance
-        rot_accel.x -= gyro.x * radians(400.0) / terminal_rotation_rate;
-        rot_accel.y -= gyro.y * radians(400.0) / terminal_rotation_rate;
-        rot_accel.z -= gyro.z * radians(400.0) / terminal_rotation_rate;
-    }
+    // Calculate linear drag forces
+    Vector3f linear_drag_forces;
+    calculate_drag_force(velocity_air_bf, frame_property.linear_drag_coefficient, linear_drag_forces);
+    // Add forces in body frame accel
+    body_accel -= linear_drag_forces / frame_property.weight;
+
+    // Calculate angular drag forces
+    Vector3f angular_drag_forces;
+    calculate_drag_force(gyro, frame_property.angular_drag_coefficient, angular_drag_forces);
+    // Add forces in body frame accel
+    body_accel -= angular_drag_forces / frame_property.weight;
+}
 
-    float terminal_velocity = 3.0;
-    if (terminal_velocity > 0) {
-        // air resistance
-        Vector3f air_resistance = -velocity_air_ef * (GRAVITY_MSS/terminal_velocity);
-        body_accel += dcm.transposed() * air_resistance;
-    }
+/**
+ * @brief
+ *
+ * @param velocity Body frame velocity of fluid
+ * @param drag_coefficient Drag coefficient of body
+ * @param force Output forces
+ * $ F_D = rho * V^2 * A * C_D / 2 $
+ * rho = water density (kg/m^3), V = volume (m^3), A = area (m^2), C_D = drag_coefficient
+ */
+void Submarine::calculate_drag_force(const Vector3f &velocity, const Vector3f &drag_coefficient, Vector3f &force)
+{
+    const Vector3f velocity_2(
+        abs(velocity.x) * velocity.x,
```

Comment:
please comment why you are maintaining the sign instead of squaring it properly as the equation above.